python/lib/sift_py/rule/service.py

```diff
@@ -48,35 +54,87 @@ def __init__(self, channel: SiftChannel):
         self._rule_service_stub = RuleServiceStub(channel)
 
     def load_rules_from_yaml(
-        self, paths: List[Path], sub_expressions: List[SubExpression]
+        self,
+        paths: List[Path],
+        sub_expressions: Optional[List[SubExpression]] = None,
+        channel_references: Optional[List[RuleChannelReference]] = None,
     ) -> List[RuleConfig]:
+        """
+        Loads rules from a YAML spec, and creates or updates the rules in the Sift API.
+        If the rule expression should be interpolated from sub-expressions, provide a list of `SubExpression` objects.
+        If the rule does not contain channel references in its YAML definition, provide a dict of rule names mapped
+        to a list of channel references. Otherwise if the YAML definition contains channel references, the `channel_references_map`
+        should be omitted. If channel references are present in both the YAML definition and provided in the `channel_references_map`,
+        or if neither are provided for a given rule, an exception will be thrown.
+        For more on rule YAML definitions, see `sift_py.ingestion.config.yaml.spec.RuleYamlSpec`.
+        """
         namespaced_rules = load_rule_namespaces(paths)
 
         interpolation_map: Dict[str, Dict[str, Any]] = {}
-        for sub_expression in sub_expressions:
-            interpolation_map[sub_expression.fully_qualified_rule_name] = sub_expression.expressions
+        if sub_expressions:
+            for sub_expression in sub_expressions:
+                interpolation_map[sub_expression.fully_qualified_rule_name] = (
+                    sub_expression.expressions
+                )
 
         rule_configs = []
         for namespace, rule_yamls in namespaced_rules.items():
             for rule_yaml in rule_yamls:
-                channel_references: List[
+                yaml_channel_references = rule_yaml.get("channel_references", [])
+                arg_channel_references = None
+
+                if channel_references:
+                    for channel_map in channel_references:
```

Comment:
Would `rule_channel_reference` make more sense than `channel_map` here?

Reply:
Yeah I like that!